### DISCUSSION TITLE
Set different line-length for .py and for .ipynb

### DISCUSSION THREAD

**Discussion by arogozhnikov** [astral-sh/ruff#13723] [Q&A]
Hi ruff team, I'd like to set much longer line-length for notebooks (than regular python files), but can't find the right way to do it.

There is a way to switch off max line-width rule in linter, but I want to use formatter on notebooks (with larger line-length).

**Comment by MichaReiser** [comment]
hi @arogozhnikov 

I don't think there's a way to configure this today unless your notebooks and regular python files are stored in different directories. If your notebooks are in a different directory, you can then create a `ruff.toml` in that directory, set `extend = <root_config.toml>`, and override `format.line-length`.

**Reply by arogozhnikov** [reply]
unfortunately they aren't - notebooks have some associated scripts in the same folder.

Formatting for notebooks looks very helpful, but for now I have to switch it off  :/

**Comment by aaron-siegel** [comment]
I would like to be able to do this as well (in our case, to use a shorter line length in Jupyter notebooks). It seems like a fairly obvious feature, since the user environment is so different between `.py` and `.ipynb`.

Thanks!
Aaron
